src/hex_tarball.erl

```diff
@@ -439,6 +492,76 @@ gzip_no_header(Uncompressed) ->
         zlib:close(Zstream)
     end.
 
+%%====================================================================
+%% Meta-Validation
+%%====================================================================
+-spec is_valid_meta(metadata()) -> {ok, list()} | {error, map()}.
+is_valid_meta(Metadata) ->
+    F = fun(K, Acc) -> 
+            case validate_meta(K, Metadata) of 
+                ok -> 
+                    Acc;
+                {warning, Warns} ->
+                    Warnings  = maps:get(warnings, Acc) ++ Warns,
+                    maps:put(warnings, Warnings, Acc);
+                {error, Errs} -> 
+                    Errors = maps:get(errors, Acc) ++ Errs,
+                    maps:put(errors, Errors, Acc)
+            end
+        end,
+    case lists:foldl(F, #{errors => [], warnings => []}, ?META_VALIDATIONS) of
+        #{errors := [], warnings := Warnings} -> 
+            {ok, Warnings};
+        Errors -> 
+         {error, Errors}
+    end.
+
+validate_meta(has_semver, #{<<"version">> := Ver}) -> 
+     case verl:parse(Ver) of
+         error ->
+             {error, [{<<"version">>, invalid}]};
+         _ -> 
+          ok
+     end;
+validate_meta(has_deprecated, Metadata) ->
+    Keys = maps:keys(Metadata),
+    Fields = ?DEPRECATED_META_FIELDS,
+    find_for(warning, deprecated, Fields, in, Keys);
+validate_meta(has_unknown, Metadata) ->
+    Keys = maps:keys(Metadata),
+    Fields = ?META_FIELDS ++ ?REQUIRED_META_FIELDS ++ ?DEPRECATED_META_FIELDS,
+    find_for(warning, unknown_field, Fields, not_in, Keys);
```

Comment:
This should be an error or filtered before included in the tarball.

Reply:
aye... will filter it out.

Reply:
Well hmmm... on one hand, it makes sense to ignore and warn for the sake of UX, otoh I consider it an error.